### PR TITLE
[8.0][ADD] account_invoice_merge_stock_picking_link addon

### PR DESCRIPTION
**Pull request by antespi**
# Merge invoices linked to pickings

This is a glue addon in order to keep picking links when two or more invoices
are merged
# Installation

This addons is automacally installed if 'OCA/account-invoicing/account_invoice_merge'
and 'OCA/stock-logistics-workflow' are installed

### CODE REVIEW

File: account_invoice_merge_stock_picking_link/README.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+Merge invoices linked to pickings
+=================================
+
+This is a glue addon in order to keep picking links when two or more invoices
+are merged
+
+Installation
+============
+
+This addons is automacally installed if 'OCA/account-invoicing/account_invoice_merge'
```
Comment (pedrobaeza) [inline account_invoice_merge_stock_picking_link/README.rst]: automatically

File: account_invoice_merge_stock_picking_link/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Account invoice merge with picking links",
+    "summary": "Merge invoices linked to pickings",
+    "version": "8.0.1.0.0",
+    "category": "Finance",
```
Comment (pedrobaeza) [inline account_invoice_merge_stock_picking_link/__openerp__.py]: Accounting & Finance

**Comment by pedrobaeza** [comment]
You are not keeping move_line_ids reference

File: account_invoice_merge_stock_picking_link/README.rst (hunk 1 above)
Comment (antespi) [inline account_invoice_merge_stock_picking_link/README.rst]: Cool!

File: account_invoice_merge_stock_picking_link/__openerp__.py (hunk 2 above)
Comment (antespi) [inline account_invoice_merge_stock_picking_link/__openerp__.py]: Done

**Comment by antespi** [comment]
Rebase https://github.com/OCA/account-invoicing/pull/138 and resolv conflicts

**Comment by pedrobaeza** [comment]
You have made an incorrect rebase. Please try again

**Comment by antespi** [comment]
I think now is better rebased

**Comment by antespi** [comment]
Travis error is related with `account_invoice_merge_purchase` and already in 8.0 branch

File: account_invoice_merge/models/account_invoice.py [hunk 3]
```diff
@@ -202,7 +202,8 @@ def do_merge(self, keep_references=True, date_invoice=False):
                              ('invoice_id', '=', new_invoice_id)])
                         if invoice_line_ids:
                             so_line.write(
-                                {'invoice_lines': [(6, 0, invoice_line_ids)]})
```
Comment (pedrobaeza) [inline account_invoice_merge/models/account_invoice.py]: You should rename then the variable name to `invoice_lines` to avoid misunderstandings

**Review by rafaelbn — COMMENTED** [review]
This branch has conflicts that must be resolved. Please @antespi review cc @pedrobaeza @Tecnativa

**Comment by pedrobaeza** [comment]
@luc-demeyer please review if the refactorization in account_invoice_merge needed here is correct for you.

**Comment by luc-demeyer** [comment]
based upon code review of the account_invoice_merge changes:
nice work, code is more modular and hence 👍

Two minor remarks for code readability:
1) change 'record' to 'invoice_line' in the 'make_merge_key(self, record, fields)'
2) rename '_first_invoice_line_values' method to '_new_invoice_line_values'

**Comment by antespi** [comment]
Long long time ago!

**Pull request closed without merging**